README.md

```diff
@@ -2,41 +2,50 @@
 
 This is an exporter for Prometheus to get Mesos data.
 
-The configuration, `config.json` should contain all Mesos slaves you want to
-monitor.
+## Building and running
 
-Prometheus configuration to add this exporter:
+    make
+    ./mesos_exporter <flags>
 
-    job: {
-      name: "mesos-exporter"
-      scrape_interval: "5s"
+### Flags
 
-      target_group: {
-        target: "http://mesosexporter.example.com/metrics"
-      }
-    }
+Name                       | Description
+---------------------------|------------
+web.listen-address         | Address to listen on for web interface and telemetry.
+web.telemetry-path         | Path under which to expose metrics.
+exporter.discovery         | Enable auto disovery of elected master and active slaves.
+exporter.discovery.master  | URL to a Mesos master.
+exporter.local-address     | Address to connect to the local slave if not using discovery.
+exporter.interval          | Interval at which to fetch the Mesos slave metrics.
 
-Where `http://mesosexporter.example.com/metrics` is the URL to this exporter.
 
-Building the binary can be done by running `make`.
+The mesos_exporter uses the [glog](https://godoc.org/github.com/golang/glog) library for logging. With the default
+parameters, nothing will be logged. Use `-logtostderr` to enable logging to
+stderr and `--help` to see more options about logging.
 
-Run the binary with the following command:
+### Modes
+mesos_exporter can operate in two modes: discovery and local.
+
+In local mode only the IP specified with the commandline flag `-exporter.local-address` will be queried and exported.
+This mode is to facilitate having one exporter per Mesos slave.
+
+In discovery mode the Mesos slaves are discovered using the `-exporter.discovery.master` flag. The exporter will fetch
+all slave metrics and export them. 
+This mode lets you have one exporter per Mesos cluster.
 
-    ./bin/mesos_exporter -config-file=config.json
 
 ### Docker
 
 A Docker container is available at
-https://registry.hub.docker.com/u/antonlindstrom/mesos-exporter/
+https://registry.hub.docker.com/u/prom/mesos-exporter
```

Comment:
Oh, I noticed we still need to create that on the Docker Hub before merging - /cc @discordianfish 

It'll need to be with an underscore though instead of dash, to match this repo name.